exchanges/kucoin/kucoin_test.go

```diff
@@ -0,0 +1,2419 @@
+	if err != nil {
+		t.Error("GetStopOrder() error", err)
+	}
+}
+
+func TestGetAllStopOrder(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip(credentialsNotSet)
+	}
+
+	_, err := ku.GetAllStopOrder(context.Background(), "", "", "", "", "", time.Time{}, time.Time{}, 0, 0)
+	if err != nil {
+		t.Error("GetAllStopOrder() error", err)
+	}
+}
+
+func TestGetStopOrderByClientID(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip(credentialsNotSet)
+	}
+
+	_, err := ku.GetStopOrderByClientID(context.Background(), "", "5bd6e9286d99522a52e458de")
+	if err != nil {
+		t.Error("GetStopOrderByClientID() error", err)
+	}
+}
+
+func TestCancelStopOrderByClientID(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.Skip(credentialsNotSet)
+	}
+
+	_, err := ku.CancelStopOrderByClientID(context.Background(), "", "5bd6e9286d99522a52e458de")
+	if err != nil && err.Error() != "order_not_exist_or_not_allow_to_cancel" {
+		t.Error("CancelStopOrderByClientID() error", err)
+	}
+}
+
+func TestGetAllAccounts(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip(credentialsNotSet)
+	}
+
+	_, err := ku.GetAllAccounts(context.Background(), "", "")
+	if err != nil {
+		t.Error("GetAllAccounts() error", err)
+	}
+}
+
+func TestGetAccount(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip(credentialsNotSet)
+	}
+
+	_, err := ku.GetAccount(context.Background(), "62fcd1969474ea0001fd20e4")
+	if err != nil && err.Error() != "account not exist" {
+		t.Error("GetAccount() error", err)
+	}
+}
+
+func TestGetAccountLedgers(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip(credentialsNotSet)
+	}
+	_, err := ku.GetAccountLedgers(context.Background(), "", "", "", time.Time{}, time.Time{})
+	if err != nil {
+		t.Error("GetAccountLedgers() error", err)
+	}
+}
+
+func TestGetAccountSummaryInformation(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip(credentialsNotSet)
+	}
+	if _, err := ku.GetAccountSummaryInformation(context.Background()); err != nil {
+		t.Error(err)
+	}
+}
+
+func TestGetSubAccountBalance(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip(credentialsNotSet)
+	}
+
+	_, err := ku.GetSubAccountBalance(context.Background(), "62fcd1969474ea0001fd20e4", false)
+	if err != nil && err.Error() != "User not found." {
+		t.Error("GetSubAccountBalance() error", err)
+	}
+}
+
+func TestGetAggregatedSubAccountBalance(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip(credentialsNotSet)
+	}
+	_, err := ku.GetAggregatedSubAccountBalance(context.Background())
+	if err != nil {
+		t.Error("GetAggregatedSubAccountBalance() error", err)
+	}
+}
+
+func TestGetPaginatedSubAccountInformation(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip(credentialsNotSet)
+	}
+	_, err := ku.GetPaginatedSubAccountInformation(context.Background(), 0, 10)
```

Comment:
Frustratingly, when there are no accounts, the response is `{"code":"200000","data":[]}`, so this test results in:
` GetPaginatedSubAccountInformation() error json: cannot unmarshal array into Go struct field Response.data of type kucoin.SubAccountsResponse`

Would you be able to handle that?

Reply:
May be with JSON Unmarshaler